src/main/java/uk/gov/hmcts/darts/audit/service/AuditServiceImpl.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.hmcts.darts.audit.service;
+
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import jakarta.persistence.TypedQuery;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Predicate;
+import jakarta.persistence.criteria.Root;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.darts.audit.model.AuditSearchQuery;
+import uk.gov.hmcts.darts.common.entity.Audit;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@AllArgsConstructor
+@Service
+@Slf4j
+public class AuditServiceImpl implements AuditService {
+
+    public static final String CASE_ID = "caseId";
```

Comment:
if we make every column name in src/main/java/uk/gov/hmcts/darts/common/entity/Audit.java a constant, then you don't have to define them separately here.
see DailyListEntity.java for reference.

Reply:
these are not column names but entity member variable names. I doesn't make sense to make constants out of that.